src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -200,6 +209,38 @@ class Definitions {
     def ObjectMethods = List(Object_eq, Object_ne, Object_synchronized, Object_clone,
         Object_finalize, Object_notify, Object_notifyAll, Object_wait, Object_waitL, Object_waitLI)
 
+  /** A trait with the following signature:
+   *
+   *    trait EqClass[-U] {
+   *      /** Comparison operations between values in the same equality class */
+   *      final def == [T >: this.type <: EqClass[_](other: T)(implicit ce: Eq[T]): Boolean = this.equals(other)
+   *      final def != [T >: this.type <: EqClass[_](other: T)(implicit ce: Eq[T]): Boolean = this.equals(other)
+   *    }
+   *
```

Comment:
A minor typo here: missing closing bracket for type parameters.

This seems to me a clever design that I feel a little difficult to appreciate fully the intricacies, e.g. what would happen: 

- if we change `trait EqClass[-U]` to `trait EqClass`, `U` is not explicitly used in the body?
- if we use `[T >: this.type <: EqClass[T]` instead `[T >: this.type <: EqClass[_]`?